apps/web/src/app/[locale]/challenge/[slug]/_context.tsx

```diff
@@ -0,0 +1,29 @@
+'use client';
+import { createContext, useState, useContext, type PropsWithChildren } from 'react';
+
+interface TrackVisibilityContextType {
+  isTrackTitleVisible: boolean | null;
+  setIsTrackTitleVisible: React.Dispatch<React.SetStateAction<boolean | null>>;
+}
+
+export const TrackVisibilityContext = createContext<TrackVisibilityContextType | undefined>(
```

Comment:
I think we don't need `undefined` there as we will always have those values when our components are wrapped with the provider

```suggestion
export const TrackVisibilityContext = createContext({} as TrackVisibilityContextType)
```

Reply:
oh thanks for the code i didn't know i could get rid of setting initial value like this